kubernetes/config/api/types.go

```diff
@@ -0,0 +1,204 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
```

Comment:
could you add a README to document how they are generated, what they are used for?

Reply:
updated